src/Common/tests/System/Net/VirtualNetwork/VirtualNetwork.cs

```diff
@@ -31,14 +34,15 @@ public void ReadFrame(bool server, out byte[] buffer)
                 packetQueue = _serverWriteQueue;
             }
 
-            semaphore.Wait(TestConfiguration.PassingTestTimeoutMilliseconds);
-            buffer = packetQueue.Dequeue();
+            semaphore.Wait(WaitForReadDataTimeoutMilliseconds);
+
+            Debug.Assert(packetQueue.TryDequeue(out buffer), "Packet queue: TryDequeue failed.");
```

Comment:
> This is fine, though you could use a BlockingCollection<byte[]> instead of the SemaphoreSlim + ConcurrentQueue<T>.

Thanks for pointing it out! I don't have enough time to change this now.

> Otherwise, the call to TryDequeue gets removed in release builds.

Yup, I see it failed on the Windows Ret build.